_api/hasura/cron/syncCoSouls.ts

```diff
@@ -216,24 +229,118 @@ const syncCoSoulToken = async (
   );
 };
 
-async function updateCoSoulOnChain(cosoul: CoSoul, totalPGIVE: number) {
+const syncPatchCoSoulToken = async (
+  updatedCosouls: { cosoul: CoSoul; localPGIVE: number }[]
+) => {
+  let payload = paddedHex(PGIVE_SLOT, 2, true); //1byte for slot
+  const successMessages = [];
+  for (const { cosoul, localPGIVE } of updatedCosouls) {
+    if (localPGIVE > 0) {
+      const pGIVE = Math.floor(localPGIVE);
+      //four bytes for pgive and four bytes for tokenId
+      payload += getPayload(pGIVE, cosoul.token_id);
+      successMessages.push(
+        `set PGIVE on chain for tokenId: ${cosoul.token_id} address: ${cosoul.address} to ${localPGIVE}`
+      );
+    } else {
+      console.log(
+        'skipping setting on-chain PGIVE because it is 0 for tokenId: ' +
+          cosoul.token_id +
+          ' address: ' +
+          cosoul.address
+      );
+    }
+  }
+  if (payload.length > 4) {
+    // 0x and one byte for slot 0x00
+    //if only one needs to be updated on chain, we can use setOnChainPGIVE
+    if (payload.length === 20) {
+      //0x00 plus 4 bytes for pgive and 4 bytes for tokenId
+      const tx = await setOnChainPGIVE(
```

Comment:
do we need to use this individual syncer? i think the code would be cleaner if we just always use the batch rather than all these ifs... but we can change later.

Reply:
if it's only one token to be updated I think we don't need the extra operations in batch pgive like getting the length and dividing it by 8 to decide the number of tokens to update and looping on tokens.
@crabsinger  @CryptoGraffe  if these extra operations won't cost anything I can make all of them use batch syncer